The test case fails for me with the next errors:

```
    ==283519==ERROR: AddressSanitizer: heap-use-after-free on address 0x613000016218 at pc 0x7f5c0550d843 bp 
    0x7fffbae9b4f0 sp 0x7fffbae9b4e0
    READ of size 4 at 0x613000016218 thread T0
    #0 0x7f5c0550d842 in bdb_back_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c:2771
    #1 0x7f5c054f9101 in bdb_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_ldif2db.c:385
    #2 0x7f5c054a9513 in ldbm_back_ldif2ldbm ldap/servers/slapd/back-ldbm/ldif2ldbm.c:147
    #3 0x562cfdabf836 in slapd_exemode_ldif2db ldap/servers/slapd/main.c:2100
    #4 0x7f5c0c088041 in __libc_start_main (/lib64/libc.so.6+0x27041)
    #5 0x562cfdac295d in _start (/usr/sbin/ns-slapd+0x3395d)

    0x613000016218 is located 88 bytes inside of 328-byte region [0x6130000161c0,0x613000016308)
    freed by thread T0 here:
    #0 0x7f5c0c9d6307 in __interceptor_free (/lib64/libasan.so.6+0xb0307)
    #1 0x7f5c0c67ecfa in slapi_ch_free (/usr/lib64/dirsrv/libslapd.so.0+0xe3cfa)
    #2 0x7f5c05513b0b in bdb_import_main ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c:2608
    #3 0x7f5c0550d4cf in bdb_back_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c:2768
    #4 0x7f5c054f9101 in bdb_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_ldif2db.c:385
    #5 0x7f5c054a9513 in ldbm_back_ldif2ldbm ldap/servers/slapd/back-ldbm/ldif2ldbm.c:147
    #6 0x562cfdabf836 in slapd_exemode_ldif2db ldap/servers/slapd/main.c:2100
    #7 0x7f5c0c088041 in __libc_start_main (/lib64/libc.so.6+0x27041)

    previously allocated by thread T0 here:
    #0 0x7f5c0c9d6837 in __interceptor_calloc (/lib64/libasan.so.6+0xb0837)
    #1 0x7f5c0c67e915 in slapi_ch_calloc (/usr/lib64/dirsrv/libslapd.so.0+0xe3915)
    #2 0x7f5c0550cb5a in bdb_back_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c:2647
    #3 0x7f5c054f9101 in bdb_ldif2db ldap/servers/slapd/back-ldbm/db-bdb/bdb_ldif2db.c:385
    #4 0x7f5c054a9513 in ldbm_back_ldif2ldbm ldap/servers/slapd/back-ldbm/ldif2ldbm.c:147
    #5 0x562cfdabf836 in slapd_exemode_ldif2db ldap/servers/slapd/main.c:2100
    #6 0x7f5c0c088041 in __libc_start_main (/lib64/libc.so.6+0x27041)

    SUMMARY: AddressSanitizer: heap-use-after-free ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c:2771 in bdb_back_ldif2db
    Shadow bytes around the buggy address:
      0x0c267fffabf0: fd fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
      0x0c267fffac00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
      0x0c267fffac10: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
      0x0c267fffac20: 00 00 00 00 00 00 00 00 00 00 00 00 00 fa fa fa
      0x0c267fffac30: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
    =>0x0c267fffac40: fd fd fd[fd]fd fd fd fd fd fd fd fd fd fd fd fd
      0x0c267fffac50: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
      0x0c267fffac60: fd fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
      0x0c267fffac70: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
      0x0c267fffac80: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
      0x0c267fffac90: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
    Shadow byte legend (one shadow byte represents 8 application bytes):
      Addressable:           00
      Partially addressable: 01 02 03 04 05 06 07
      Heap left redzone:       fa
      Freed heap region:       fd
      Stack left redzone:      f1
      Stack mid redzone:       f2
      Stack right redzone:     f3
      Stack after return:      f5
      Stack use after scope:   f8
      Global redzone:          f9
      Global init order:       f6
      Poisoned by user:        f7
      Container overflow:      fc
      Array cookie:            ac
      Intra object redzone:    bb
      ASan internal:           fe
      Left alloca redzone:     ca
      Right alloca redzone:    cb
      Shadow gap:              cc
    ==283519==ABORTING
    FAILED

    =================================== FAILURES ====================================
    ___________________________ test_ldif2db_syntax_check ___________________________

    topo = <lib389.topologies.TopologyMain object at 0x7f985d6bc8e0>

    def test_ldif2db_syntax_check(topo):
        """ldif2db should return a warning when a skipped entry has occured.
        :id: 85e75670-42c5-4062-9edc-7f117c97a06f
        :setup:
            1. Standalone Instance
            2. Ldif entry that violates syntax check rule (empty givenname)
        :steps:
            1. Create an ldif file which violates the syntax checking rule
            2. Stop the server and import ldif file with ldif2db
        :expected results:
            1. ldif2db import returns a warning to signify skipped entries
        """
        import_ldif1 = _create_syntax_err_ldif(topo)
        # Import the offending LDIF data - offline
        topo.standalone.stop()
        ret = topo.standalone.ldif2db('userRoot', None, None, None, import_ldif1)
    >       assert ret == SKIPDENTRY
    E       assert False == 2
    E         -False
    E         +2
```

--- a/dirsrvtests/tests/suites/import/import_test.py
+++ b/dirsrvtests/tests/suites/import/import_test.py
@@ -15,7 +15,7 @@ import pytest
 import time
 import glob
 from lib389.topologies import topology_st as topo
-from lib389._constants import DEFAULT_SUFFIX
+from lib389._constants import DEFAULT_SUFFIX, TaskWarning
 from lib389.dbgen import dbgen_users
 from lib389.tasks import ImportTask
 from lib389.index import Indexes
@@ -139,6 +139,38 @@ def _create_bogus_ldif(topo):
     return import_ldif1
 
 
+def _create_syntax_err_ldif(topo):
+    """
+    Create an incorrect ldif entry that violates syntax check
+    """
+    ldif_dir = topo.standalone.get_ldif_dir()
+    line1 = """dn: dc=example,dc=com
+objectClass: top
+objectClass: domain
+dc: example
+dn: ou=groups,dc=example,dc=com
+objectClass: top
+objectClass: organizationalUnit
+ou: groups
+dn: uid=JHunt,ou=groups,dc=example,dc=com
+objectClass: top
+objectClass: person
+objectClass: organizationalPerson
+objectClass: inetOrgPerson
+objectclass: inetUser
+cn: James Hunt
+sn: Hunt
+uid: JHunt
+givenName:
+"""
+    with open(f'{ldif_dir}/syntax_err.ldif', 'w') as out:
+        out.write(f'{line1}')
+        os.chmod(out.name, 0o777)
+    out.close()
+    import_ldif1 = ldif_dir + '/syntax_err.ldif'
+    return import_ldif1
+
+
 def test_import_with_index(topo, _import_clean):
     """
     Add an index, then import via cn=tasks
@@ -211,6 +243,26 @@ def test_ldif2db_allows_entries_without_a_parent_to_be_imported(topo, _import_cl
     # which violates schema, ending line
     topo.standalone.searchErrorsLog('import_producer - import userRoot: Skipping entry '
                                     '"dc=example,dc=com" which violates schema')
+    topo.standalone.start()
+
+
+def test_ldif2db_syntax_check(topo):
+    """ldif2db should return a warning when a skipped entry has occured.
+    :id: 85e75670-42c5-4062-9edc-7f117c97a06f
+    :setup:
+        1. Standalone Instance
+        2. Ldif entry that violates syntax check rule (empty givenname)
+    :steps:
+        1. Create an ldif file which violates the syntax checking rule
+        2. Stop the server and import ldif file with ldif2db
+    :expected results:
+        1. ldif2db import returns a warning to signify skipped entries
+    """
+    import_ldif1 = _create_syntax_err_ldif(topo)
+    # Import the offending LDIF data - offline
+    topo.standalone.stop()
+    ret = topo.standalone.ldif2db('userRoot', None, None, None, import_ldif1)
+    assert ret == TaskWarning.WARN_SKIPPED_IMPORT_ENTRY
     topo.standalone.start()
 
 

--- a/ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c
+++ b/ldap/servers/slapd/back-ldbm/db-bdb/bdb_import.c
@@ -2563,7 +2563,7 @@ error:
                 slapi_task_dec_refcount(job->task);
             }
             import_all_done(job, ret);
-            ret = 1;
+            ret |= WARN_UPGARDE_DN_FORMAT_ALL;
         } else if (NEED_DN_NORM == ret) {
             import_log_notice(job, SLAPI_LOG_NOTICE, "bdb_import_main",
                               "%s complete. %s needs upgradednformat.",
@@ -2572,7 +2572,7 @@ error:
                 slapi_task_dec_refcount(job->task);
             }
             import_all_done(job, ret);
-            ret = 2;
+            ret |= WARN_UPGRADE_DN_FORMAT;
         } else if (NEED_DN_NORM_SP == ret) {
             import_log_notice(job, SLAPI_LOG_NOTICE, "bdb_import_main",
                               "%s complete. %s needs upgradednformat spaces.",
@@ -2581,7 +2581,7 @@ error:
                 slapi_task_dec_refcount(job->task);
             }
             import_all_done(job, ret);
-            ret = 3;
+            ret |= WARN_UPGRADE_DN_FORMAT_SPACE;
         } else {
             ret = -1;
             if (job->task != NULL) {
@@ -2598,6 +2598,11 @@ error:
             slapi_task_dec_refcount(job->task);
         }
         import_all_done(job, ret);
+    }
+
+    /* set task warning if there are no errors */
+    if((!ret) && (job->skipped)) {
+        ret |= WARN_SKIPPED_IMPORT_ENTRY;
     }
 
     /* This instance isn't busy anymore */
@@ -2637,6 +2642,7 @@ bdb_back_ldif2db(Slapi_PBlock *pb)
     int total_files, i;
     int up_flags = 0;
     PRThread *thread = NULL;
+    int ret = 0;
 
     slapi_pblock_get(pb, SLAPI_BACKEND, &be);
     if (be == NULL) {
@@ -2764,7 +2770,15 @@ bdb_back_ldif2db(Slapi_PBlock *pb)
     }
 
     /* old style -- do it all synchronously (THIS IS GOING AWAY SOON) */
-    return import_main_offline((void *)job);
+    ret = import_main_offline((void *)job);
+
+    /* no error just warning, reset ret */
+    if(ret &= WARN_SKIPPED_IMPORT_ENTRY) {
+        slapi_pblock_set_task_warning(pb, WARN_SKIPPED_IMPORT_ENTRY);
+        ret = 0;
+    }
+
+    return ret;
 }
 
 struct _import_merge_thang

--- a/ldap/servers/slapd/main.c
+++ b/ldap/servers/slapd/main.c
@@ -2104,6 +2104,14 @@ slapd_exemode_ldif2db(struct main_config *mcfg)
                       plugin->plg_name);
         return_value = -1;
     }
+
+    /* check for task warnings */
+    if(!return_value) {
+        if((return_value = slapi_pblock_get_task_warning(pb))) {
+            slapi_log_err(SLAPI_LOG_INFO, "slapd_exemode_ldif2db","returning task warning: %d\n", return_value);
+        }
+    }
+
     slapi_pblock_destroy(pb);
     charray_free(instances);
     charray_free(mcfg->cmd_line_instance_names);

--- a/ldap/servers/slapd/pblock.c
+++ b/ldap/servers/slapd/pblock.c
@@ -28,11 +28,13 @@
 #define SLAPI_LDIF_DUMP_REPLICA 2003
 #define SLAPI_PWDPOLICY 2004
 #define SLAPI_PW_ENTRY 2005
+#define SLAPI_TASK_WARNING 2006
 
 /* Used for checking assertions about pblocks in some cases. */
 #define SLAPI_HINT 9999
 
 static PRLock *pblock_analytics_lock = NULL;
+
 
 static PLHashNumber
 hash_int_func(const void *key)
@@ -4287,6 +4289,28 @@ slapi_pblock_set_ldif_dump_replica(Slapi_PBlock *pb, int32_t dump_replica)
 #endif
     _pblock_assert_pb_task(pb);
     pb->pb_task->ldif_dump_replica = dump_replica;
+}
+
+int32_t
+slapi_pblock_get_task_warning(Slapi_PBlock *pb)
+{
+#ifdef PBLOCK_ANALYTICS
+    pblock_analytics_record(pb, SLAPI_TASK_WARNING);
+#endif
+    if (pb->pb_task != NULL) {
+        return pb->pb_task->task_warning;
+    }
+    return 0;
+}
+
+void
+slapi_pblock_set_task_warning(Slapi_PBlock *pb, task_warning warning)
+{
+#ifdef PBLOCK_ANALYTICS
+    pblock_analytics_record(pb, SLAPI_TASK_WARNING);
+#endif
+    _pblock_assert_pb_task(pb);
+    pb->pb_task->task_warning = warning;
 }
 
 void *

--- a/ldap/servers/slapd/pblock_v3.h
+++ b/ldap/servers/slapd/pblock_v3.h
@@ -68,6 +68,7 @@ typedef struct _slapi_pblock_task
     int ldif2db_noattrindexes;
     int ldif_printkey;
     int task_flags;
+    int32_t task_warning;
     int import_state;
 
     int server_running; /* indicate that server is running */

--- a/ldap/servers/slapd/slapi-private.h
+++ b/ldap/servers/slapd/slapi-private.h
@@ -1464,6 +1464,17 @@ void slapi_pblock_set_operation_notes(Slapi_PBlock *pb, uint32_t opnotes);
 void slapi_pblock_set_flag_operation_notes(Slapi_PBlock *pb, uint32_t opflag);
 void slapi_pblock_set_result_text_if_empty(Slapi_PBlock *pb, char *text);
 
+/* task warnings */
+typedef enum task_warning_t{
+    WARN_UPGARDE_DN_FORMAT_ALL    = (1 << 0),
+    WARN_UPGRADE_DN_FORMAT        = (1 << 1),
+    WARN_UPGRADE_DN_FORMAT_SPACE  = (1 << 2),
+    WARN_SKIPPED_IMPORT_ENTRY     = (1 << 3)
+} task_warning;
+
+int32_t slapi_pblock_get_task_warning(Slapi_PBlock *pb);
+void slapi_pblock_set_task_warning(Slapi_PBlock *pb, task_warning warn);
+
 
 int slapi_exists_or_add_internal(Slapi_DN *dn, const char *filter, const char *entry, const char *modifier_name);
 

--- a/src/lib389/lib389/__init__.py
+++ b/src/lib389/lib389/__init__.py
@@ -2673,14 +2673,14 @@ class DirSrv(SimpleLDAPObject, object):
         try:
             result = subprocess.check_output(cmd, encoding='utf-8')
         except subprocess.CalledProcessError as e:
-            self.log.debug("Command: %s failed with the return code %s and the error %s",
-                           format_cmd_list(cmd), e.returncode, e.output)
-            return False
-
-        self.log.debug("ldif2db output: BEGIN")
-        for line in result.split("\n"):
-            self.log.debug(line)
-        self.log.debug("ldif2db output: END")
+            if e.returncode == TaskWarning.WARN_SKIPPED_IMPORT_ENTRY:
+                self.log.debug("Command: %s skipped import entry warning %s",
+                               format_cmd_list(cmd), e.returncode)
+                return e.returncode
+            else:
+                self.log.debug("Command: %s failed with the return code %s and the error %s",
+                               format_cmd_list(cmd), e.returncode, e.output)
+                return False
 
         return True
 

--- a/src/lib389/lib389/_constants.py
+++ b/src/lib389/lib389/_constants.py
@@ -166,6 +166,13 @@ DB2BAK = 'db2bak'
 DB2INDEX = 'db2index'
 DBSCAN = 'dbscan'
 
+# Task warnings
+class TaskWarning(IntEnum):
+    WARN_UPGARDE_DN_FORMAT_ALL      = (1 << 0)
+    WARN_UPGRADE_DN_FORMAT          = (1 << 1)
+    WARN_UPGRADE_DN_FORMAT_SPACE    = (1 << 2)
+    WARN_SKIPPED_IMPORT_ENTRY       = (1 << 3)
+
 RDN_REPLICA = "cn=replica"
 
 RETROCL_SUFFIX = "cn=changelog"

--- a/src/lib389/lib389/cli_ctl/dbtasks.py
+++ b/src/lib389/lib389/cli_ctl/dbtasks.py
@@ -7,6 +7,7 @@
 # See LICENSE for details.
 # --- END COPYRIGHT BLOCK ---
 
+from lib389._constants import TaskWarning
 
 def dbtasks_db2index(inst, log, args):
     if not inst.db2index(bename=args.backend):
@@ -44,10 +45,13 @@ def dbtasks_db2ldif(inst, log, args):
 
 
 def dbtasks_ldif2db(inst, log, args):
-    if not inst.ldif2db(bename=args.backend, encrypt=args.encrypted, import_file=args.ldif,
-                        suffixes=None, excludeSuffixes=None, import_cl=False):
+    ret = inst.ldif2db(bename=args.backend, encrypt=args.encrypted, import_file=args.ldif,
+                        suffixes=None, excludeSuffixes=None, import_cl=False)
+    if not ret:
         log.fatal("ldif2db failed")
         return False
+    elif ret == TaskWarning.WARN_SKIPPED_IMPORT_ENTRY:
+        log.warn("ldif2db successful with skipped entries")
     else:
         log.info("ldif2db successful")
 

